src/app/components/Filters/FieldsetDate.jsx

```diff
@@ -1,27 +1,108 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import Actions from '../../actions/Actions.js';
+import {
+  extend as _extend,
+} from 'underscore';
+import NumberFormat from 'react-number-format';
+
+class FieldsetDate extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      dateAfter: '0',
+      dateBefore: '0',
+      selectedFacets: this.props.selectedFacets,
+    };
+
+    this.inputChange = this.inputChange.bind(this);
+  }
+
+  /**
+   * inputChange()
+   * Updates the states and the store based on the input values.
+   *
+   */
+  inputChange(e) {
+    const value = e.target.value;
+    let displayValue = '';
+
+    if (e.target.name === 'start-date') {
+      displayValue = 'dateAfter';
+    } else if (e.target.name === 'end-date') {
+      displayValue = 'dateBefore';
+    }
+
+    if (displayValue) {
+      this.setState(
+        {
+          [displayValue]: value,
+          selectedFacets: _extend(
+            this.state.selectedFacets,
+            { [displayValue]: value }
+          ),
+        },
+        () => {
+          Actions.updateSelectedFacets(this.state.selectedFacets);
+        }
+      );
+    }
+  }
 
-class FilterPopup extends React.Component {
   render() {
+    let errorMessage = '';
+
+    if (this.state.dateAfter && this.state.dateBefore) {
+      if (Number(this.state.dateBefore) < Number(this.state.dateAfter)) {
+        errorMessage = 'end year should be later than start year.';
+      }
+    } else {
+      errorMessage = '';
+    }
+
     return (
       <fieldset>
         <legend>Date</legend>
-        <div>
-          <label htmlFor="start-date">Start Year
-            <input id="start-date" name="start-date" className="form-text" type="date" />
+        <div id="input-container">
+          <label htmlFor="startDate" id="startDate-label">Start Year
```

Comment:
The `htmlFor` attribute points to `startDate` but the `input` id is `start-date`. They should match. The `aria-labelledby` is correct and the `input` is inside the `label`, but for completeness the `htmlFor` should match.